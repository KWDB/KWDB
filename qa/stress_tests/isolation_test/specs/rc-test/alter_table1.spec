setup
{
 CREATE TABLE a (i int PRIMARY KEY);
 CREATE TABLE b (a_id int);
 CREATE INDEX idx1 on b(a_id);
 INSERT INTO a VALUES (0), (1), (2), (3);
 INSERT INTO b SELECT generate_series(1,1000) % 4;
}

teardown
{
 DROP TABLE a, b;
}

session s1
step s1		{ BEGIN transaction isolation level read committed; }
step at1	{ ALTER TABLE b ADD CONSTRAINT bfk FOREIGN KEY (a_id) REFERENCES a (i) NOT VALID; }
step sc1	{ COMMIT; }
step s2		{ BEGIN; }
step at2	{ ALTER TABLE b VALIDATE CONSTRAINT bfk; }
step sc2	{ COMMIT; }

session s2
setup		{ BEGIN transaction isolation level read committed; }
step rx1	{ SELECT * FROM b WHERE a_id = 1 LIMIT 1; }
step wx		{ INSERT INTO b VALUES (0); }
step rx3	{ SELECT * FROM b WHERE a_id = 3 LIMIT 3; }
step c2		{ COMMIT; }

permutation s1 at1 sc1 s2 at2 sc2 rx1 wx rx3 c2
permutation s1 at1 sc1 s2 at2 rx1 sc2 wx rx3 c2
permutation s1 at1 sc1 s2 at2 rx1 wx sc2 rx3 c2
permutation s1 at1 sc1 s2 at2 rx1 wx rx3 sc2 c2
permutation rx1 wx s1 rx3 c2 at1 sc1 s2 at2 sc2
permutation rx1 wx rx3 s1 at1 c2 sc1 s2 at2 sc2
permutation rx1 wx rx3 s1 c2 at1 sc1 s2 at2 sc2
permutation rx1 wx rx3 c2 s1 at1 sc1 s2 at2 sc2
