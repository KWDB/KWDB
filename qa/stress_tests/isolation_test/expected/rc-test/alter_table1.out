Parsed test spec with 2 sessions

starting permutation: s1 at1 sc1 s2 at2 sc2 rx1 wx rx3 c2
step s1: BEGIN transaction isolation level read committed;  finished.
step at1: ALTER TABLE b ADD CONSTRAINT bfk FOREIGN KEY (a_id) REFERENCES a (i) NOT VALID;  finished.
ERROR:  can not use multi-statement transactions involving a schema change under weak isolation levels
step sc1: COMMIT;  finished.
step s2: BEGIN;  finished.
step at2: ALTER TABLE b VALIDATE CONSTRAINT bfk;  finished.
ERROR:  constraint "bfk" does not exist on table b
step sc2: COMMIT;  finished.
step rx1: SELECT * FROM b WHERE a_id = 1 LIMIT 1;  finished.
a_id
----
   1
(1 row)

step wx: INSERT INTO b VALUES (0);  finished.
step rx3: SELECT * FROM b WHERE a_id = 3 LIMIT 3;  finished.
a_id
----
   3
   3
   3
(3 rows)

step c2: COMMIT;  finished.

starting permutation: s1 at1 sc1 s2 at2 rx1 sc2 wx rx3 c2
step s1: BEGIN transaction isolation level read committed;  finished.
step at1: ALTER TABLE b ADD CONSTRAINT bfk FOREIGN KEY (a_id) REFERENCES a (i) NOT VALID;  finished.
ERROR:  can not use multi-statement transactions involving a schema change under weak isolation levels
step sc1: COMMIT;  finished.
step s2: BEGIN;  finished.
step at2: ALTER TABLE b VALIDATE CONSTRAINT bfk;  finished.
ERROR:  constraint "bfk" does not exist on table b
step rx1: SELECT * FROM b WHERE a_id = 1 LIMIT 1;  finished.
a_id
----
   1
(1 row)

step sc2: COMMIT;  finished.
step wx: INSERT INTO b VALUES (0);  finished.
step rx3: SELECT * FROM b WHERE a_id = 3 LIMIT 3;  finished.
a_id
----
   3
   3
   3
(3 rows)

step c2: COMMIT;  finished.

starting permutation: s1 at1 sc1 s2 at2 rx1 wx sc2 rx3 c2
step s1: BEGIN transaction isolation level read committed;  finished.
step at1: ALTER TABLE b ADD CONSTRAINT bfk FOREIGN KEY (a_id) REFERENCES a (i) NOT VALID;  finished.
ERROR:  can not use multi-statement transactions involving a schema change under weak isolation levels
step sc1: COMMIT;  finished.
step s2: BEGIN;  finished.
step at2: ALTER TABLE b VALIDATE CONSTRAINT bfk;  finished.
ERROR:  constraint "bfk" does not exist on table b
step rx1: SELECT * FROM b WHERE a_id = 1 LIMIT 1;  finished.
a_id
----
   1
(1 row)

step wx: INSERT INTO b VALUES (0);  finished.
step sc2: COMMIT;  finished.
step rx3: SELECT * FROM b WHERE a_id = 3 LIMIT 3;  finished.
a_id
----
   3
   3
   3
(3 rows)

step c2: COMMIT;  finished.

starting permutation: s1 at1 sc1 s2 at2 rx1 wx rx3 sc2 c2
step s1: BEGIN transaction isolation level read committed;  finished.
step at1: ALTER TABLE b ADD CONSTRAINT bfk FOREIGN KEY (a_id) REFERENCES a (i) NOT VALID;  finished.
ERROR:  can not use multi-statement transactions involving a schema change under weak isolation levels
step sc1: COMMIT;  finished.
step s2: BEGIN;  finished.
step at2: ALTER TABLE b VALIDATE CONSTRAINT bfk;  finished.
ERROR:  constraint "bfk" does not exist on table b
step rx1: SELECT * FROM b WHERE a_id = 1 LIMIT 1;  finished.
a_id
----
   1
(1 row)

step wx: INSERT INTO b VALUES (0);  finished.
step rx3: SELECT * FROM b WHERE a_id = 3 LIMIT 3;  finished.
a_id
----
   3
   3
   3
(3 rows)

step sc2: COMMIT;  finished.
step c2: COMMIT;  finished.

starting permutation: rx1 wx s1 rx3 c2 at1 sc1 s2 at2 sc2
step rx1: SELECT * FROM b WHERE a_id = 1 LIMIT 1;  finished.
a_id
----
   1
(1 row)

step wx: INSERT INTO b VALUES (0);  finished.
step s1: BEGIN transaction isolation level read committed;  finished.
step rx3: SELECT * FROM b WHERE a_id = 3 LIMIT 3;  finished.
a_id
----
   3
   3
   3
(3 rows)

step c2: COMMIT;  finished.
step at1: ALTER TABLE b ADD CONSTRAINT bfk FOREIGN KEY (a_id) REFERENCES a (i) NOT VALID;  finished.
ERROR:  can not use multi-statement transactions involving a schema change under weak isolation levels
step sc1: COMMIT;  finished.
step s2: BEGIN;  finished.
step at2: ALTER TABLE b VALIDATE CONSTRAINT bfk;  finished.
ERROR:  constraint "bfk" does not exist on table b
step sc2: COMMIT;  finished.

starting permutation: rx1 wx rx3 s1 at1 c2 sc1 s2 at2 sc2
step rx1: SELECT * FROM b WHERE a_id = 1 LIMIT 1;  finished.
a_id
----
   1
(1 row)

step wx: INSERT INTO b VALUES (0);  finished.
step rx3: SELECT * FROM b WHERE a_id = 3 LIMIT 3;  finished.
a_id
----
   3
   3
   3
(3 rows)

step s1: BEGIN transaction isolation level read committed;  finished.
step at1: ALTER TABLE b ADD CONSTRAINT bfk FOREIGN KEY (a_id) REFERENCES a (i) NOT VALID;  finished.
ERROR:  can not use multi-statement transactions involving a schema change under weak isolation levels
step c2: COMMIT;  finished.
step sc1: COMMIT;  finished.
step s2: BEGIN;  finished.
step at2: ALTER TABLE b VALIDATE CONSTRAINT bfk;  finished.
ERROR:  constraint "bfk" does not exist on table b
step sc2: COMMIT;  finished.

starting permutation: rx1 wx rx3 s1 c2 at1 sc1 s2 at2 sc2
step rx1: SELECT * FROM b WHERE a_id = 1 LIMIT 1;  finished.
a_id
----
   1
(1 row)

step wx: INSERT INTO b VALUES (0);  finished.
step rx3: SELECT * FROM b WHERE a_id = 3 LIMIT 3;  finished.
a_id
----
   3
   3
   3
(3 rows)

step s1: BEGIN transaction isolation level read committed;  finished.
step c2: COMMIT;  finished.
step at1: ALTER TABLE b ADD CONSTRAINT bfk FOREIGN KEY (a_id) REFERENCES a (i) NOT VALID;  finished.
ERROR:  can not use multi-statement transactions involving a schema change under weak isolation levels
step sc1: COMMIT;  finished.
step s2: BEGIN;  finished.
step at2: ALTER TABLE b VALIDATE CONSTRAINT bfk;  finished.
ERROR:  constraint "bfk" does not exist on table b
step sc2: COMMIT;  finished.

starting permutation: rx1 wx rx3 c2 s1 at1 sc1 s2 at2 sc2
step rx1: SELECT * FROM b WHERE a_id = 1 LIMIT 1;  finished.
a_id
----
   1
(1 row)

step wx: INSERT INTO b VALUES (0);  finished.
step rx3: SELECT * FROM b WHERE a_id = 3 LIMIT 3;  finished.
a_id
----
   3
   3
   3
(3 rows)

step c2: COMMIT;  finished.
step s1: BEGIN transaction isolation level read committed;  finished.
step at1: ALTER TABLE b ADD CONSTRAINT bfk FOREIGN KEY (a_id) REFERENCES a (i) NOT VALID;  finished.
ERROR:  can not use multi-statement transactions involving a schema change under weak isolation levels
step sc1: COMMIT;  finished.
step s2: BEGIN;  finished.
step at2: ALTER TABLE b VALIDATE CONSTRAINT bfk;  finished.
ERROR:  constraint "bfk" does not exist on table b
step sc2: COMMIT;  finished.
